### PR TITLE
Remove msleep in owl_scan_routine

### PR DESCRIPTION
**Pull request by rickywu0421**
To remove msleep, I introduce a method used by broadcom FullMAC
driver: set a timer in scan routine, and once the timer timeout, we
start informing "dummy" BSS to kernel, then call cfg80211_scan_done()
to finish scan. The detail is in github issue#19.

### CODE REVIEW

**Comment by rickywu0421** [comment]
I'm not sure the method is suitable for virtual driver, because we don't really have to "scan" a BSS.

File: vwifi.c [hunk 1]
```diff
@@ -244,6 +235,35 @@ static void owl_scan_routine(struct work_struct *w)
     mutex_unlock(&owl->lock);
 }
 
+/* Callback called when the scan timer timeout. This function just schedule
```
Comment (jserv) [inline vwifi.c:238]: Improve the grammar.

File: vwifi.c [hunk 2]
```diff
@@ -244,6 +235,35 @@ static void owl_scan_routine(struct work_struct *w)
...
+ */
+static void owl_scan_timeout(struct timer_list *t)
+{
+    struct owl_context *owl = container_of(t, struct owl_context, scan_timeout);
+
+    if (owl->scan_request)
+        schedule_work(&owl->ws_scan_timeout);
+}
+
+/* "Scan routine". It simulate a fake BSS scan (In fact, do nothing), and set
```
Comment (jserv) [inline vwifi.c:249]: Ditto. Take care of the grammar.

**Comment by jserv** [comment]
Thank @rickywu0421 for contributing. I amended the commit message.

**Pull request merged**